table.lua

```diff
@@ -518,30 +545,28 @@ end
 --- local spliced = table.splice(arr, 3, 7) -- {30, 40, 50, 60, 70}
 --- log(serpent.line(arr)) -- {10, 20, 80, 90} (values were removed)
 --- ```
---- @param arr array
+--- @param array array
 --- @param start int default: `1`
---- @param stop? int Stop at this index. If zero or negative, will stop `n` items from the end of the array (default: `#arr`).
---- @return array A new array with the extracted values.
-function flib_table.splice(arr, start, stop)
-  local output = {}
-  local n = #arr
+--- @param stop? int Stop at this index. If zero or negative, will stop `n` items from the end of the array (default: `#array`).
+--- @return array #A new array with the extracted values.
+function flib_table.splice(array, start, stop)
+  local n = #array
 
   start = start or 1
-  stop = stop or n
-  stop = stop <= 0 and (n + stop) or stop
-
   if start < 1 or start > n then
     return {}
   end
 
+  stop = stop or n
```

Comment:
Reflowed the function to do evaluations and table creation after fail state. Micro perf improvment and less GC churn for fail state.